docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,245 @@
+```
+
+Creating applications from multiple sources would allow users to configure multiple services stored at various sources within the same application.
+
+## Motivation
+
+The main motivation behind this enhancement proposal is to allow users to create an application using services that are stored in various sources.
+
+### Goals
+
+The goals of the enhancement are:
+
+#### **Supporting multiple sources for creating an application**
+
+Users should be able to specify multiple sources to add services to the application. Argo CD should compile all the sources and reconcile each source individually for creating the application.
+
+#### **Allow specifying external value files for Helm repositories**
+
+Users should be able to specify different sources for Helm charts and values files. The Helm charts specified by the user could be available in Git or Helm repository and the value files are stored in Git. Argo CD should track changes in both the Helm charts and the value files repository and reconcile the application.
+
+#### Changes to UI
+
+The UI should allow users to add multiple sources while creating the application. For each source, UI should allow users to add multiple external values files Helm projects. We would need a separate proposal for changes to UI.
+
+#### Changes to cli
+
+The cli would need to support adding a list of resources instead of just one while creating the application. `values` field should allow referencing value files from other sources. We would need a separate proposal for changes to cli.
+
+### Non-goals
+* 
+
+## Proposal
+
+### Add new `sources` field in Application Spec
+
+The proposal is to add a new field `sources` which would allow users to input list of `ApplicationSource`s. We would mark field `source` as deprecated and would ignore the details under `source` with details under `sources` field. 
+
+Below example shows how the yaml would look like for `source` and `sources` field. We would ignore the `source` field and apply the resources mentioned under `sources` field.
+
+```yaml
+spec:
+  source:
+    repoURL: https://github.com/elastic/helm-charts/tree/main/elasticsearch
+    targetRevision: 6.8
+    helm:
+      valueFiles:
+        - values.yaml
+  sources:                                          # new field
+    - repoURL: https://github.com/helm/charts
+      targetRevision: master
+      path: incubator/elasticsearch
+      helm:
+        valueFiles:
+          - values.yaml
+```
+
+### Make `path/chart` field optional
+
+While adding sources to the application, users can decide not to add `path/chart` field in the application yaml. The controller will not generate manifests for the sources that do not have `path/chart` field specified. For example, in the below application spec, controller will generate the manifest for `elasticsearch` source but not for source `my-repo`.
```

Comment:
If I recall correctly, `path` is currently optional and defaults to `.`. Since the meaning of `path: ""` or path-field-absent is changing, we'll need to document that in any migration documentation.

Reply:
The `path` is actually required. Right now the user must provide either `path` or `chart` field. So making it optional is not a breaking change.